### PR TITLE
Fix domain derivation on LC syncAggregate validation

### PR DESCRIPTION
**Pull request by dapplion**
**Motivation**

- https://github.com/ethereum/consensus-specs/pull/3284

**Description**

Fix off by one error causing invalid signature errors on fork boundary

### CODE REVIEW

File: packages/light-client/src/spec/validateLightClientUpdate.ts [hunk 1]
```diff
@@ -113,7 +113,7 @@ export function validateLightClientUpdate(
 
   const signingRoot = ssz.phase0.SigningData.hashTreeRoot({
     objectRoot: ssz.phase0.BeaconBlockHeader.hashTreeRoot(update.attestedHeader.beacon),
-    domain: store.config.getDomain(update.signatureSlot, DOMAIN_SYNC_COMMITTEE),
+    domain: store.config.getDomain(update.signatureSlot - 1, DOMAIN_SYNC_COMMITTEE),
```
Comment (g11tech) [inline packages/light-client/src/spec/validateLightClientUpdate.ts:116]: or should we use attestedHeader.beacon.slot?

Comment (g11tech) [reply packages/light-client/src/spec/validateLightClientUpdate.ts:116]: ahh just checked the linked PR , lgtm :+1:

**Review by g11tech — APPROVED** [review]

File: packages/light-client/src/spec/validateLightClientUpdate.ts (hunk 1 above)
Comment (etan-status) [reply packages/light-client/src/spec/validateLightClientUpdate.ts:116]: sadly, attested_header.beacon.slot may be incorrect.

Example:

BELLATRIX_FORK_EPOCH = 0
CAPELLA_FORK_EPOCH = 255
DENEB_FORK_EPOCH = 256

attested_header.slot = 8159 (bellatrix)
signature_slot = 8192 (Deneb)
fork_version_slot = 8191 (Capella)

so, sync committee would be determined by sync committee at slot 8192, the fork version is determined by slot 8191, and the object is from slot 8159

Comment (g11tech) [reply packages/light-client/src/spec/validateLightClientUpdate.ts:116]: thanks @etan-status for the insightful example :pray:  :heart:

**Comment by wemeetagain** [comment]
:tada: This PR is included in [v1.7.0](https://github.com/ChainSafe/lodestar/releases/tag/v1.7.0) :tada:

**Pull request merged**